pyanaconda/modules/storage/devicetree/viewer.py

```diff
@@ -462,13 +462,61 @@ def _get_os_data(self, root):
         }
         return data
 
+    def _get_mount_point_constraints_data(self, spec):
+        """Get the mount point data.
+
+        :param spec: an instance of PartSpec
+        :return: an instance of MountPointConstraintsData
+        """
+        data = MountPointConstraintsData()
+        data.mount_point = spec.mountpoint or ""
+        data.required_filesystem_type = spec.fstype or ""
+        data.encryption_allowed = spec.encrypted
+        data.logical_volume_allowed = spec.lv
+
+        return data
+
+    def get_mount_point_constraints(self):
+        """Get list of constraints on mountpoints for the current platform
+
+        Also provides hints if the partition is required or recommended.
+
+        This includes mount points required to boot (e.g. /boot/efi, /boot)
+        and the / partition which is always considered to be required.
+
+        /boot is not required in general but can be required in some cases,
+        depending on the filesystem on the root partition (ie crypted root).
+
+        :return: a list of mount points with its constraints
+        """
+
+        constraints = []
+
+        # Root partition is required
+        root_partition = PartSpec(mountpoint="/", lv=True, thin=True, encrypted=True)
+        root_constraint = self._get_mount_point_constraints_data(root_partition)
+        root_constraint.required = True
+        constraints.append(root_constraint)
+
+        # Platform partitions are required except for /boot partiotion which is recommended
+        for p in platform.partitions:
+            if p.mountpoint:
+                constraint = self._get_mount_point_constraints_data(p)
+                if p.mountpoint == "/boot":
+                    constraint.recommended = True
```

Comment:
I wonder if this shouldn't be actually defined in the `Storage Constraints` section of our configuration files. That section already contains other constraints like this one and it might be useful for Web UI and Cockpit Storage.

Reply:
It makes sense to me, but I hesitated to use configuration files perhaps because of the outlook of more dynamic API (like /boot being required depending on fs on /). Nevertheless the static information (/boot (a mount point) required or recommended in general) can still be in the configuration file.

Reply:
Something like
```
boot_partition_recommended = True
```
or even (not sure if it makes sense in this form)
```
recommended_mount_points = /boot
required_mount_points = /
```
Well, `required_mount_points` seems to be covered by existing `must_not_be_on_root`
Maybe we should wait for more use cases / requirements on required.
Actually, `recommended_mount_points` might become useful also for /home - seems like a good setting.

Reply:
@poncovka would you be OK with handling it as a follow-up? We are dealing with test flakiness on webui part and it could be handy to have this merged.

Reply:
Do you want an issue about considering this followup? If this can be handled this year, I would not create one.

Reply:
Sure, it is definitely just a suggestion. I didn't want to block this pull request.

Reply:
Yes, please create the issue.